addons/hr_holidays/models/hr_accrual_plan.py

```diff
@@ -0,0 +1,244 @@
+    start_delta = fields.Datetime(compute="_compute_start_delta", store=True)
+    # Accrue of
+    added_hours = fields.Float("Hours per worked days", required=True, help="The number of hours that will be incremented for every period")
+    # Accrual period
+    frequency = fields.Selection([
+        ('daily', 'daily'),
+        ('weekly', 'weekly'),
+        ('twice a month', 'twice a month'),
+        ('monthly', 'monthly'),
+        ('twice a year', 'twice a year'),
+        ('quarterly', 'quarterly'),
+        ('yearly', 'yearly'),
+        ('anniversary', 'anniversary'),
+    ], default='daily', required=True)
+    start_of_period = fields.Selection([
+        ('first', 'on the first day'),
+        ('last', 'on the last day'),
+        ('first monday', 'on the first monday'),
+        ('last friday', 'on the last friday')
+    ], default='first')
+    # Max accrual
+    maximum = fields.Float('Maximum accrual hours', help="The maximum allocated hours. The hours above this limit will not be added", default="1000")
+    maximum_type = fields.Selection([
+        ('hours', 'hours'),
+        ('days', 'days'),
+    ], default='hours')
+    maximum_period = fields.Selection([
+        ('period', 'period'),
+        ('days','days'),
+        ('weeks', 'weeks'),
+        ('months', 'months'),
+        ('years', 'years'),
+    ], default='days')
+
+    @api.depends('start_delta', 'start_count', 'start_type')
+    def _compute_start_delta(self):
+        year_zero = datetime(1, 1, 1)
+        for record in self :
+            delta = relativedelta(days=0)
+            if record.start_type == 'days':
+                delta = relativedelta(days=record.start_count)
+            if record.start_type == 'months':
+                delta = relativedelta(months=record.start_count)
+            if record.start_type == 'years':
+                delta = relativedelta(years=record.start_count)
+            record.start_delta = year_zero + delta
+
+    def _get_start_after_delta(self):
+        delta = relativedelta(days=0)
+        if self.start_type == 'days':
+            delta = relativedelta(days=self.start_count)
+        if self.start_type == 'months':
+            delta = relativedelta(months=self.start_count)
+        if self.start_type == 'years':
+            delta = relativedelta(years=self.start_count)
+        return delta
+
+    def _get_period(self, employee):
+        # Find the first day of the next and previous periods
+        period ={}
+        next = False
+        previous = False
+        today = datetime.combine(fields.Date.today(), time(0, 0, 0))
+        start_date = datetime.combine(employee._get_date_start_work(), time(0, 0, 0))
+
+        if self.frequency == 'daily':
+            next = today + relativedelta(days=1)
+            previous = today
+
+        if self.frequency == 'weekly':
+            if self.start_of_period == "first monday" or self.start_of_period == "first":
```

Comment:
I would also replace the spaces with underscores in the selection fields.

```suggestion
            if self.start_of_period in ["first monday", "first"]:
```